lib/routes/try.js

```diff
@@ -2,6 +2,30 @@ var express = require('express');
 var request = require('request');
 var streamBuffers = require("stream-buffers");
 
+function getRequestedUrl( req ) {
+  var url = req.headers['x-request-url'];
+  if (!url) {
+    return resp.status(500).send('URL must be provided in x-request-url header');
+  }
+  delete req.headers['x-request-url'];
+  return url;
+}
+
+function serializeHeaders( req ) {
+  var array = [];
+  Object.keys(req.headers).forEach( function( key ) {
+    var value = req.headers[key];
+    array.push( key + ': ' + value );
+  });
+  return array.join('\n');
+}
+
+function enrichResponseWithRequestHeaders( req, resp ) {
+  var serializedHeaders = serializeHeaders( req );
```

Comment:
not 100% sure what's going on here - is this just echoing back incoming headers as another header on a response? wouldn't the client know what headers have been sent. I'm probably missing all kinds context!